Makefile

```diff
@@ -1,12 +1,18 @@
+# Give Selfie information about the operating system ($OS is defined on Windows only)
+ifndef OS
+	# use uname or fail with "Unknown"
+	OS := $(shell sh -c 'uname 2>/dev/null || echo Unknown')
```

Comment:
This is nice but in the end I think the code you used before was better. How about going back to the code before but fail if the OS cannot be determined to be one of the three supported OSs? Pass to selfie just integers, as before.

Reply:
done